packages/template-retail-react-app/app/hooks/use-shopper-context-search-params.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {
+    useUsid,
+    useShopperContext,
+    useShopperContextsMutation
+} from '@salesforce/commerce-sdk-react'
+import {isServer, isHydrated} from '@salesforce/retail-react-app/app/utils/utils'
+import {useQueryClient} from '@tanstack/react-query'
+
+// Constants
+import {
+    SHOPPER_CONTEXT_SEARCH_PARAMS,
+    SHOPPER_CONTEXT_CUSTOM_QUALIFIERS_SEARCH_PARAMS,
+    SHOPPER_CONTEXT_ASSIGNMENT_QUALIFIERS_SEARCH_PARAMS,
+    SHOPPER_CONTEXT_ARRAY_FIELDS
+} from '@salesforce/retail-react-app/app/constants'
+
+/*
+ * This hook will set the shopper context when search params pertinant
+ * to shopper context are present.
+ */
+export const useShopperContextSearchParams = (siteId) => {
+    const {usid} = useUsid()
+    const queryClient = useQueryClient()
+    const {search} = useLocation()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {parameters: {usid, siteId}},
+        {enabled: !isServer}
+    )
```

Comment:
When there is no shopper context this returns a 404 error in the console. Should we hide logging that error? 
<img width="1286" alt="Screenshot 2024-09-06 at 11 51 56 AM" src="https://github.com/user-attachments/assets/7ff86510-cb67-4d53-a538-043b91f5353d">